map_model/src/objects/building.rs

```diff
@@ -227,110 +227,193 @@ fn sidewalk_to_bike(sidewalk_pos: Position, map: &Map) -> Option<(Position, Posi
+            AmenityType::BikeStore => vec!["bicycle"],
+            AmenityType::CarRepair => vec!["car_repair"],
+            AmenityType::CarShare => vec!["car_sharing"],
+            AmenityType::Childcare => vec!["childcare", "kindergarten"],
+            AmenityType::ConvenienceStore => vec!["convenience"],
+            AmenityType::Culture => vec![
+                "arts_centre",
+                "art_gallery",
+                "cinema",
+                "library",
+                "museum",
+                "theatre",
+            ],
+            AmenityType::Financial => vec!["bank"],
             AmenityType::Food => vec![
                 "restaurant",
                 "cafe",
+                "farm",
                 "fast_food",
                 "food_court",
                 "ice_cream",
                 "pastry",
+                "pasta",
+                "spices",
+                "seafood",
+                "tea",
+                "coffee",
+                "cheese",
+                "chocolate",
                 "deli",
-                "greengrocer",
                 "bakery",
                 "butcher",
                 "confectionery",
                 "beverages",
                 "alcohol",
             ],
-            AmenityType::Bar => vec!["bar", "lounge", "pub", "nightclub"],
+            AmenityType::Laundry => vec!["dry_cleaning", "laundry", "tailor"],
             AmenityType::Medical => vec![
                 "chiropractor",
                 "clinic",
                 "dentist",
                 "hospital",
                 "pharmacy",
+                "doctors",
                 "optician",
             ],
-            AmenityType::Religious => vec!["place_of_worship"],
-            AmenityType::Education => vec!["college", "school", "university"],
-            AmenityType::Financial => vec!["bank"],
+            AmenityType::Pet => vec!["veterinary", "pet", "animal_boarding", "pet_grooming"],
             AmenityType::PostOffice => vec!["post_office"],
-            AmenityType::Culture => vec![
-                "arts_centre",
-                "art_gallery",
-                "cinema",
-                "library",
-                "museum",
-                "theatre",
-            ],
-            AmenityType::Childcare => vec!["childcare", "kindergarten"],
+            AmenityType::Religious => vec!["place_of_worship", "religion"],
+            AmenityType::School => vec!["school"],
             AmenityType::Shopping => vec![
+                "wholesale",
+                "bag",
+                "marketplace",
                 "second_hand",
+                "charity",
                 "clothes",
+                "lottery",
                 "furniture",
                 "shoes",
+                "mall",
                 "department_store",
                 "car",
+                "tailor",
+                "nutrition_supplements",
+                "watches",
+                "craft",
+                "wool",
```

Comment:
Cool to see us handling more tags. :+1:

Some of these seem quite obscure - there's such a long tail of tags in OSM, that I don't think adding every single one makes sense, so I'm curious how you're choosing which ones to add?

Reply:
Yes I certainly noticed the long tail. Some of these could be trimmed out and not make much difference, I was trying to go on the side of more is better. There is probably no good way to pick and choose unless maybe we do a specialty shop category to put more of the bespoke types of retailers?

Reply:
@matkoniecz has helpfully shared https://taginfo.openstreetmap.org/keys/amenity#values before, which exactly shows frequency of use. I think including more is fine for now. We could also consider making a catch-all `Other` category. It depends what we want to do with these categories. The immediate consequence to breaking these down into finer groups is increasing the column height in the 15m tool, and presenting the user with much more nuance up-front.

Reply:
Just in case you weren't already aware of the possibility, one methodical way would be to prioritize based on frequency of tag use — e.g. all the values associated with "shop=*":
https://taginfo.openstreetmap.org/keys/shop#values

Reply:
(whoops, didn't see @dabreegster's comment when I posted)... same same.

Reply:
Looking at the frequency of shop and amenity it does look like a threshold could work so we are somewhat systematic about ignoring the tails. Keeping anything > 0.05% would still keep a lot of the specialty shops but cut out the really obscure ones. Shopping would still be quite a big category though but maybe that's not so terrible.